src/pages/RegisterPage.tsx

```diff
@@ -13,22 +13,27 @@ type Inputs = {
     email: string,
     name: string,
     surname: string,
+    phone?: string | null | undefined,
     password: string,
     confirm_password: string,
     terms: boolean,
 }
+
+const phoneNumberRegex = /^(\+\d{1,3}[- ]?)?(\d{3}[- ]\d{3}[- ]\d{3}|\d{10})$/;
+
 const schema = yup.object().shape({
     email: yup.string().email().required(),
     name: yup.string().min(2).max(50).required(),
     surname: yup.string().min(2).max(50).required(),
+    phone: yup.string().nullable().transform((value, originalValue) => originalValue === '' ? null : value).matches(phoneNumberRegex, 'Phone is not valid').notRequired(),
```

Comment:
wystarczy nie dawac `.required()` i juz jest not required :D ale ogolnie to nie blad, widac chociaz co sie dzieje :)